Doc/c-api/contextvars.rst

```diff
@@ -101,21 +101,76 @@ Context object management functions:
    current context for the current thread.  Returns ``0`` on success,
    and ``-1`` on error.
 
-.. c:function:: int PyContext_AddWatcher(PyContext_WatchCallback callback)
-
-   Register *callback* as a context object watcher for the current interpreter.
-   Return an ID which may be passed to :c:func:`PyContext_ClearWatcher`.
-   In case of error (e.g. no more watcher IDs available),
-   return ``-1`` and set an exception.
+.. c:function:: int PyContext_AddWatcher(PyObject *callback)
+
+   Registers the callable object *callback* as a context object watcher for the
+   current interpreter.  When a context event occurs, *callback* is called with
+   two arguments:
+
+   #. An event type ID from :c:type:`PyContextEvent`.
+   #. An object containing event-specific supplemental data; see
+      :c:type:`PyContextEvent` for details.
```

Comment:
I think my preference would be to create a new `contextvars.ContextSwitchedEvent` class that holds the Context object and pass that instead, but this is a smaller change.